Functions/Mock.ps1

```diff
@@ -1540,28 +1544,65 @@ function Repair-ConflictingParameters {
     param(
         [Parameter(Mandatory = $true)]
         [System.Management.Automation.CommandMetadata]
-        $Metadata
+        $Metadata,
+        [Parameter()]
+        [string[]]
+        $RemoveParameterType,
+        [Parameter()]
+        [string[]]
+        $RemoveParameterValidation
     )
 
+    $repairedMetadata = New-Object System.Management.Automation.CommandMetadata -ArgumentList $Metadata
     $paramMetadatas = @()
-    $paramMetadatas += $Metadata.Parameters.Values
+    $paramMetadatas += $repairedMetadata.Parameters.Values
+
+    $conflictingParams = Get-ConflictingParameterNames
 
     foreach ($paramMetadata in $paramMetadatas) {
         if ($paramMetadata.IsDynamic) {
             continue
         }
 
-        $conflictingParams = Get-ConflictingParameterNames
         if ($conflictingParams -contains $paramMetadata.Name) {
             $paramName = $paramMetadata.Name
             $newName = "_$paramName"
             $paramMetadata.Name = $newName
             $paramMetadata.Aliases.Add($paramName)
 
-            $null = $Metadata.Parameters.Remove($paramName)
-            $Metadata.Parameters.Add($newName, $paramMetadata)
+            $null = $repairedMetadata.Parameters.Remove($paramName)
+            $repairedMetadata.Parameters.Add($newName, $paramMetadata)
+        }
+
+        $attrIndexesToRemove = New-Object System.Collections.ArrayList
+
+        if ($paramMetadata.Name -in $RemoveParameterType) {
+            $paramMetadata.ParameterType = [object]
+
+            for ($i = 0; $i -lt $paramMetadata.Attributes.Count; $i++) {
+                $attr = $paramMetadata.Attributes[$i]
+                if ($attr -is [PSTypeNameAttribute]) {
```

Comment:
This removal is required, otherwise PowerShell won't be able to match the new `[object]` type definition for the parameter in the corresponding parameter set.